### PR TITLE
Ticket crud

### PR DESCRIPTION
**Pull request by OlgaKochneva**
Add crud functionality for ticket.

### CODE REVIEW

File: .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<component name="ArtifactManager">
```
Comment (vakhonin) [inline .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml]: AAAAA why .idea in the pull request???

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+             ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS  ")) {
+            while (rs.next()) {
+                Long id = rs.getLong("id");
+                Long price = rs.getLong("initialPrice");
+                LocalDateTime departureDate = rs.getTimestamp("departureDate").toLocalDateTime();
+                LocalDateTime arrivalDate = rs.getTimestamp("arrivalDate").toLocalDateTime();
+                Long plane_id = rs.getLong("plane_id");
+                Plane plane = planeDAO.getById(plane_id);
+                Boolean isDeleted = rs.getBoolean("isDeleted");
+                if (!isDeleted)
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: remove it. You can add deleted option in the query

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                    flights.add(Flight.builder().id(id).initialPrice(price).plane(plane).
+                            departureDate(departureDate).arrivalDate(arrivalDate).build());
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: builder style

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        LocalDateTime arrivalDate = flight.getArrivalDate();
+        Long plane_id = flight.getPlane().getId();
+
+        String sql = String.format("INSERT INTO Flight (initialPrice, plane_id, departureDate,arrivalDate,isDeleted)" +
+                " VALUES (%d,%d,'%s','%s','%s')", price, plane_id, DateFormatter.format(depatureDate), DateFormatter.format(arrivalDate), "FALSE");
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS");
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: you can get id from executing method. Nastya did it yesterday

File: src/main/java/com/epam/lowcost/DAO/TicketDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        ticket = null;
+        return ticket;
+    }
+
+
+    private User extractUserFromRS(ResultSet rs) throws SQLException {
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/DAO/TicketDAOImpl.java]: it is used in another class. Why you duplicate code?

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+@Controller
+@RequestMapping(value = "/flight")
+public class FlightController {
+    @Autowired
+    FlightService flightService;
+
+    @Autowired
+    PlaneService planeService;
+
+    @GetMapping(value = "/getAllFlights")
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: please check url from teammate's pull requests

File: src/main/java/com/epam/lowcost/controller/FlightController.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        model.addAttribute("flight", flightService.getFlightById(id));
+        return "flight";
+    }
+
+    @PostMapping
+    public String addNewFlight(@RequestParam Map<String, String> params, Model model) {
+        //System.out.println("here");
+        //System.out.println(Long.valueOf(params.get("plane_id")));
+        //System.out.println(planeService);
+        //System.out.println("here");
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/FlightController.java]: I think these comments aren't necessary

File: src/main/java/com/epam/lowcost/service/TicketServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    public Ticket deleteTicket(long id) {
+        return ticketDAO.deleteTicket(id);
+    }
+
+    private long countPrice(long initialPrice, Ticket ticket){
+        if(ticket.isBusiness())
+            initialPrice+=5000;
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/service/TicketServiceImpl.java]: TicketService knows about additional money for some options. It is not right. Is it clear ?

File: .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml (hunk 1 above)
Comment (OlgaKochneva) [reply .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml]: i don't know, what is a problem. gitignore ignored all folders except .idea. Do you know, what could it be?

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java (hunk 2 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: it is not mine, I use Nastya's branch before you reviewed it. I merge her updated branch and resolve conflicts.

File: .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml (hunk 1 above)
Comment (OlgaKochneva) [reply .idea/artifacts/Gradle___test___test_1_0_SNAPSHOT_war__exploded_.xml]: Done

File: src/main/java/com/epam/lowcost/DAO/TicketDAOImpl.java (hunk 5 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/DAO/TicketDAOImpl.java]: Remove all duplications of this code, not only in my class and merge with Nastya's updated branch.

File: src/main/java/com/epam/lowcost/service/TicketServiceImpl.java (hunk 8 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/service/TicketServiceImpl.java]: It was test-realisation of logic. I removed this method, now ticket has initialPrice cost. We discuss correct logic and add this method later.

Comment (vakhonin) [reply src/main/java/com/epam/lowcost/service/TicketServiceImpl.java]: ok

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java (hunk 3 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: done

File: src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java (hunk 4 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java]: If Nastya did it, i have updated it today, by merging her branch. So it is done

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 6 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: done

File: src/main/java/com/epam/lowcost/controller/FlightController.java (hunk 7 above)
Comment (OlgaKochneva) [reply src/main/java/com/epam/lowcost/controller/FlightController.java]: done

**Pull request merged**